### DISCUSSION TITLE
Update title

### DISCUSSION THREAD

**Discussion by JacobHamelMottiez** [aidenlx/obsidian-zotero#301] [Help]
Hi, 

Thanks for this awesome plugin! I was wondering if when updating a note we could also update the file name title. In my case, the file name is the citekey in Zotero. However, this citekey can change over time (e.g. from Bilge_2022 to Bilge_2022a). Is there a way to make this work? I would be very grateful for any help! 

Best,

Jacob

**Comment by aidenlx** [comment]
Not planned. v2 runs the `filename` template when a note is created and never renames the file afterwards, and that is deliberate rather than a gap — ZotLit identifies a note by the `zotero-key` field in its frontmatter, not by its name, so the filename is yours to own once the note exists. Renaming user files on a metadata refresh is the kind of thing that should not happen behind your back.

The practical upshot for your case: because identity lives in frontmatter, you can rename the file yourself whenever a citekey changes. Obsidian updates the links pointing at it, and ZotLit keeps tracking the note under its new name — updates, **Open Literature Note in Obsidian**, and the citekey quick switcher all keep working.

v2 beta is available now: https://zotlit.aidenlx.site/docs/install-zotlit
Migration guide: https://zotlit.aidenlx.site/docs/how-to/migrate-from-v1
